File: project/yeonwoo-onboarding/src/main/java/com/innercircle/yeonwoo_onboarding/repository/SurveyItemRepository.java
package com.innercircle.yeonwoo_onboarding.repository;

import com.innercircle.yeonwoo_onboarding.domain.SurveyItem;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.List;

public interface SurveyItemRepository extends JpaRepository<SurveyItem, String> {
    List<SurveyItem> findBySurveyId(String surveyId);
}

File: project/sangkyeong-onboarding/domain/src/main/kotlin/org/survey/domain/survey/repository/SurveyItemRepository.kt
package org.survey.domain.survey.repository

import org.survey.domain.survey.model.SurveyItem

interface SurveyItemRepository {
    fun save(surveyItem: SurveyItem): Long

    fun saveAll(surveyItems: List<SurveyItem>): List<SurveyItem>

    fun findBySurveyId(surveyId: Long): List<SurveyItem>

    fun findAllById(surveyItemIds: List<Long>): List<SurveyItem>

    fun deleteItems(surveyItems: List<SurveyItem>)

    fun updateItems(surveyItems: List<SurveyItem>)
}
